src/Elements/Button/Button.tsx

```diff
@@ -0,0 +1,97 @@
+import cx from 'classnames';
+import * as React from 'react';
+
+//Style
+import './button.scss';
+
+// types
+interface IProps {
+  /**
+   * class attribute of the component
+   */
+  className?: string;
+  /**
+   * content of the component
+   */
+  children?: React.ReactNode;
+  /**
+   * declare if the button is visible or not
+   */
+  visible?: boolean;
+  /**
+   * icon to be shown on the button
+   */
+  icon?: string;
+  /**
+   * icon positions
+   */
+  iconPosition?: 'left' | 'right';
+  /**
+   * if the button is disabled or not
+   */
+  disabled?: boolean;
+  /**
+   * label of the component
+   */
+  label?: string;
+  /**
+   * position of the label
+   */
+  labelPosition?: 'right' | 'left'
+  /**
+   * on click event handler function
+   */
+  onClick?: Function;
+  /**
+   * the size of the button
+   */
+  size?: 'small' | 'medium' | 'large';
+  /**
+   * color of the button
+   */
+  color?: string
+}
+
+export default class Button extends React.Component<IProps, {}> {
+
+  handleClick = (e: React.MouseEvent) => {
+    const { disabled, onClick } = this.props;
+    if (disabled) {
```

Comment:
I don't think we need this check here, this function should not be invoked at all if the button is disabled. Please correct me if I am wrong.

Reply:
Because the click is on the div we can do two things: move the click from the div to the button and we can remove the id OR leave as is

Reply:
I think it's better to have the click handler on the button